src/expr.cpp

```diff
@@ -6710,9 +6738,9 @@ static llvm::Value *lUniformValueToVarying(FunctionEmitContext *ctx, llvm::Value
 
     // Otherwise we must have a uniform atomic or pointer type, so smear
     // its value across the vector lanes.
-    if (CastType<AtomicType>(type)) {
-        return lTypeConvAtomic(ctx, value, CastType<AtomicType>(type->GetAsVaryingType()), CastType<AtomicType>(type),
-                               pos);
+    if (CastType<AtomicType>(type) && CastType<AtomicType>(type->GetAsVaryingType())) {
```

Comment:
I understand that this change is to silence the static code checker, but this was unexpected to see.
I'm fine leaving it as is, but we should consider using asserts for such things.